commercetools-java-client-core/src/main/java/io/sphere/sdk/client/SolutionInfoService.java

```diff
@@ -21,9 +20,20 @@ public static synchronized SolutionInfoService getInstance() {
     }
 
     public List<SolutionInfo> getSolutionInfos() {
-        final List<SolutionInfo> solutions = new LinkedList<>();
-        loader.iterator()
-                .forEachRemaining(solution -> solutions.add(solution));
+        List<SolutionInfo> solutions  ;
```

Comment:
We could have initialized the list here and this would have allowed us to make it final.

Reply:
i am afraid that if an error happens during insertion we end up with a corrupt list 😢